client/tx_watcher.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/rs/zerolog/log"
+)
+
+type client interface {
+	TransactionByHash(ctx context.Context, hash common.Hash) (tx *types.Transaction, isPending bool, err error)
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
```

Comment:
Looks like it is just `hash` in other places.
```suggestion
	TransactionReceipt(ctx context.Context, hash common.Hash) (*types.Receipt, error)
```